### PR TITLE
NPM Path integration

### LINKED ISSUE #3: Use the npm-run module
Consider using the npm-run module to automatically add the locally installed node_modules on the path.

### PR DESCRIPTION
**Pull request by steelbrain**
- [x] Add eslint linting
- [x] Implement support
- [x] Document it in the readme
- [x] Document it in the changelog
- [x] Add specs for it

Fixes #3

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,13 +5,17 @@
...
-    "compile": "ucompiler go"
+    "compile": "ucompiler go",
+    "lint": "eslint ."
   },
   "devDependencies": {
-    "ucompiler": "^3.1.7",
-    "ucompiler-plugin-babel": "^3.1.1",
-    "ucompiler-plugin-newline": "^3.0.0",
-    "babel-preset-steelbrain": "^2.0.0"
+    "eslint": "^2.4.0",
```
Comment (Arcanemagus) [inline package.json]: Why not 2.5.3?

File: package.json [hunk 2]
```diff
@@ -5,13 +5,17 @@
...
+    "babel-eslint": "^6.0.0",
+    "babel-preset-steelbrain": "^2.0.1",
+    "eslint-config-airbnb": "^6.0.2",
```
Comment (Arcanemagus) [inline package.json]: 6.2.0

File: package.json [hunk 3]
```diff
@@ -5,13 +5,17 @@
...
+    "eslint-plugin-react": "^4.2.1",
```
Comment (Arcanemagus) [inline package.json]: 4.2.3

File: spec/.eslintrc.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{
+  "globals": {
```
Comment (Arcanemagus) [inline spec/.eslintrc.json]: Almost all of this is replaceable with:
```json
{
  "env": {
    "jasmine": true
  }
}
```

The only global you need to specify then is `waitsForPromise`, which is an Atom extension on top of Jasmine.

File: spec/main-spec.js [hunk 5]
```diff
@@ -64,20 +64,42 @@ describe('exec', function() {
...
     expect(result).toBe('')
   })
+
+  it('supports executing modules from local paths', async function() {
+    const result = await exec('sb-exec-test', [], { local: {
+      directory: Path.join(__dirname, 'fixtures', 'path')
+    } })
+    expect(result).toBe('HEY')
+  })
+  it('supports prepend', async function() {
```
Comment (Arcanemagus) [inline spec/main-spec.js]: You have spaces between the other `it` blocks, they should be down here too.

**Comment by steelbrain** [comment]
Thanks for your review Landon. The versions are not up-to-date because I copy pasted the deps section, but since it's semver shouldn't really be a problem. You are right about the spacing and the env, I'll take care of it in a follow up PR

File: README.md [hunk 6]
```diff
@@ -17,7 +17,11 @@ type $Options = {
   stream?: 'stdout' | 'stderr'  | 'both',
   env: Object,
   stdin?: ?string,
-  throwOnStderr: boolean
+  local?: {
```
Comment (Arcanemagus) [inline README.md]: This should have some sort of description as to what this is doing... from what I can tell from the code this is appending `node_module` directories into the path?

Comment (steelbrain) [inline README.md]: The description is here https://github.com/steelbrain/exec/issues/3#issuecomment-202989901

Maybe we should add another section to README explaining the config types

File: src/helpers.js [hunk 7]
```diff
@@ -34,4 +39,34 @@ export function validate(filePath: string, parameters: Array<string>, options: E
...
+  const spawnOptions = assign({}, options, {
+    env: assign(await getEnv(), options.env)
+  })
+  const local = options.local
+  if (local) {
+    const npmPath = await getNpmPath.async(local.directory)
+    for (const key in spawnOptions.env) {
+      if (spawnOptions.env.hasOwnProperty(key) && PATH_REGEX.test(key)) {
+        const value = spawnOptions.env[key]
+        spawnOptions.env[key] = local.prepend ? npmPath + PATH_SEPARATOR + value : value + PATH_SEPARATOR + npmPath
```
Comment (Arcanemagus) [inline src/helpers.js]: Should there be a check that `npmPath` actually _has_ something in there before blindly adding it? What about a check that the directory actually exists?

On Windows the environment space is actually rather limited, so creating a gigantic `PATH` variable can cause issues with path's "disappearing".

File: README.md (hunk 6 above)
Comment (Arcanemagus) [inline README.md]: That was my point exactly, the description shouldn't be buried in an issue comment, it should be in the README where your users can see what the option actually does.

**Pull request merged**